### ISSUE TITLE
One billion large Transformer training best practices

### ISSUE THREAD

**Issue by davidecaroselli** [tensorflow/tensor2tensor#722]
### Description

I launched a training of tensor2tensor Transformer with a 1.1 billion source words training set. This is the command I run on a 8 GPUs machine (NVIDIA K80):

```bash
t2t-trainer --t2t_usr_dir path/to/my/problem \
  --data_dir=$DATA_DIR \
  --problems=my_problem \
  --model=transformer \
  --hparams_set=transformer_big \
  --output_dir=$TRAIN_DIR \
  --local_eval_frequency=5000 \
  --train_steps=5000000 \
  --worker_gpu=8 \
  --hparams='batch_size=1536'
```

This is the loss from Tensorboard:
![schermata 2018-04-17 alle 19 27 07](https://user-images.githubusercontent.com/1674891/38886163-720fe20e-4275-11e8-865e-4532fa10f0e1.png)

So to me it seems that the training is proceeding well; the problem is that partial results (last checkpoint eval) are not so great in terms of BLEU score.

To me it seems that, after ~10 hours of training, the training is already at the step in which it has (more or less) converged. First of all: am I right?
If this is the case, the problem is that, with 8 GPUs and a batch size of 1536 the process has processed roughly 10% of the training set, it's too early to converge!

Have you any experience on this kind of large training? Have you any hints on how to properly setup a training like this?

### *TensorFlow* and *tensor2tensor* versions
Tensorflow: 1.7.0
tensor2tensor: 1.5.5

Thank you in advance for your help!

**Comment by mehmedes** [comment]
I wouldn't say it has more or less converged, it's rather progressing very slowly. You have 8 gpus, each processing a batch of 1500 tokens, which results in an effective batch size of 8 x 1500 tokens. In the paper they use an effective batch size of around 30k tokens! for training on the WMT corpus with about 5M sentences. Maybe you can increase your batch size using Adafactor and limiting max_length, #444.

**Comment by davidecaroselli** [comment]
@mehmedes you were right! After 2 day of training the model is still improving, so it has not converged at all!
